griptape/drivers/vector/opensearch_vector_store_driver.py

```diff
@@ -0,0 +1,184 @@
+    index_name: str = field(kw_only=True)
+    dimension: int = field(default=3, kw_only=True)
+
+    client: OpenSearch = field(default=Factory(
+        lambda self: OpenSearch(
+            hosts=[{'host': self.host, 'port': self.port}],
+            http_auth=self.http_auth,
+            use_ssl=self.use_ssl,
+            verify_certs=self.verify_certs,
+            connection_class=RequestsHttpConnection
+        ),
+        takes_self=True
+    ))
+
+    def upsert_vector(
+            self,
+            vector: list[float],
+            vector_id: Optional[str] = None,
+            namespace: Optional[str] = None,
+            meta: Optional[dict] = None,
+            **kwargs
+    ) -> str:
+
+        vector_id = vector_id if vector_id else utils.str_to_hash(str(vector))
+        doc = {
+            "vector": vector,
+            "namespace": namespace,
+            "metadata": meta
+        }
+        doc.update(kwargs)
+        response = self.client.index(index=self.index_name, id=vector_id, body=doc)
+
+        return response["_id"]
+
+    def load_entry(self, vector_id: str, namespace: Optional[str] = None) -> Optional[BaseVectorStoreDriver.Entry]:
+        try:
+            response = self.client.get(index=self.index_name, id=vector_id)
+            if response["found"]:
+                vector_data = response["_source"]
+                entry = BaseVectorStoreDriver.Entry(
+                    id=vector_id,
+                    meta=vector_data.get("metadata"),
+                    vector=vector_data.get("vector"),
+                    namespace=vector_data.get("namespace")
+                )
+                return entry
+            return None
+        except Exception as e:
+            print(f"Error while loading entry: {e}")
+            return None
+
+    def load_entries(self, namespace: Optional[str] = None) -> list[BaseVectorStoreDriver.Entry]:
+
+        query_body = {
+            "size": 10000,
+            "query": {
+                "match_all": {}
+            }
+        }
+
+        # If a namespace is provided, adjust the query to match vectors in that namespace
+        if namespace:
+            query_body["query"] = {
+                "match": {
+                    "namespace": namespace
+                }
+            }
+
+        response = self.client.search(index=self.index_name, body=query_body)
+
+        entries = []
+        for hit in response["hits"]["hits"]:
+            vector_data = hit["_source"]
+            entry = BaseVectorStoreDriver.Entry(
+                id=hit["_id"],
+                vector=vector_data.get("vector"),
+                meta=vector_data.get("metadata"),
+                namespace=vector_data.get("namespace")
+            )
+            entries.append(entry)
+
+        return entries
+
+    def query(
+            self,
+            query_vector: list[float],
+            count: Optional[int] = None,
+            field_name: str = "vector",
```

Comment:
Looks like this argument is never used. Remove it?

Reply:
We are using it in the query

Reply:
Oops, my bad.